### DISCUSSION TITLE
Runtime API to expose log message metadata

### DISCUSSION THREAD

**Discussion by madhukaw** [ballerina-platform/ballerina-standard-library#772] [General]
In the current Log API design, the Log message doesn’t contain the exact code line the log message is printed. It is useful if this information could be added to every log message.

At the moment, the only way to extract this information is through the stack trace. However, this is not a proper solution and has an adverse performance impact. It would be better if there was a Runtime API to support this requirement.

@warunalakshitha, @daneshk WDYT.

**Comment by warunalakshitha** [comment]
ATM we are supporting to get the previous module and IMO we can extend that API to retrieve the previous call info as well( ex line number).
@MadhukaHarith92  Do we have a issue for this?

**Comment by madhukaw** [comment]
Created an issue in ballerina-lang: https://github.com/ballerina-platform/ballerina-lang/issues/28247
Created an issue in log module: https://github.com/ballerina-platform/ballerina-standard-library/issues/900
